client/src/utils/DeleteModal.js

```diff
@@ -0,0 +1,52 @@
+import React, { useContext } from 'react'
+import PostContext from '../context/post/PostContext'
+
+const DeleteModal = () => {
+  const postContext = useContext(PostContext)
+  const {
+    deleteAction: { showModal, id, contentType },
+    clearDeleteModal,
+    deletePost,
+    deleteComment,
+  } = postContext
+
+  const deleteContent = (e) => {
+    e.preventDefault()
+    if (contentType === 'post') {
+      deletePost(id)
+    } else if (contentType === 'comment') {
+      deleteComment(id)
+    }
+    clearDeleteModal()
+  }
+
+  const returnModal = () => {
+    return showModal ? (
+      <div className='modal-overlay'>
+        <div className='delete-modal bg-light mx-auto text-center py-4 border border-secondary rounded'>
+          <p>Are you sure you want to delete this {contentType}?</p>
```

Comment:
is {contentType} being rendered properly? idk if it gets interpolated like this but lmk if im wrong lol

Reply:
Yes since I have it set as string, it comes out fine.